Dynavity/Dynavity/model/canvas/Canvas.swift

```diff
@@ -85,3 +77,34 @@ extension Canvas {
         umlConnectors.remove(at: index)
     }
 }
+
+extension Canvas: Codable {
+    private enum CodingKeys: String, CodingKey {
+        case canvasElements, umlConnectors, name
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        func getOrDefault<T: Decodable>(type: T.Type, key: CodingKeys, orElse: T) -> T {
+            (try? container.decode(type, forKey: key)) ?? orElse
+        }
+        let wrappedElements = getOrDefault(type: [TypeWrappedCanvasElement].self,
+                                           key: .canvasElements,
+                                           orElse: [])
+        self.canvasElements = wrappedElements.map { $0.data }
+        self.umlConnectors = getOrDefault(type: [UmlConnector].self,
+                                          key: .umlConnectors,
+                                          orElse: [])
+        self.name = try container.decode(String.self, forKey: .name)
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        let wrappedElements = canvasElements
+            .filter { !($0 is TestElement) } // TODO remove when testelements are gone
```

Comment:
I think we can delete the test elements in this PR; they've outlived their usefulness.